spikes/pikepdf-write/write_tagged_pdf.py

```
@@ -218,35 +218,65 @@ def write_tagged_pdf(
-            parent_tree_nums.append(struct_parent_idx)
-            parent_tree_nums.append(pdf.make_indirect(mcid_to_elem))
+            #    Only pages that produce MCID assignments need /StructParents.
+            if assignments:
+                max_mcid = -1
+                for mcid, zone_idx in assignments:
+                    if 0 <= zone_idx < len(zone_elems):
+                        zone_elems[zone_idx].K.append(mcid)
+                        max_mcid = max(max_mcid, mcid)
+
+                mcid_to_elem = pikepdf.Array()
+                for mcid in range(max_mcid + 1):
```

Ruff (0.15.12)

[warning] Loop control variable `mcid` not used within loop body

(B007)